sos/report/plugins/shmcli.py

```diff
@@ -0,0 +1,112 @@
+            'list physical enclosures',
+            'list failed drives'
+        ]
+
+        for subcmd in subcmds:
+            self.add_cmd_output(
+                "%s %s" % (cmd, subcmd),
+                suggest_filename="shmcli_%s" % (subcmd))
+
+        models = []
+
+        # Get the storage hardware models
+        result = self.exec_cmd('lsscsi -g')
+        if result['status'] == 0:
+            for line in result['output'].splitlines():
+                if (line.split()[2] == 'DELL'):
+                    models.append(line.split()[3])
+        models = list(set(models))
+
+        subcmds = [
+            'list emms',
+            'list drawers',
+            'list emm slots',
+            'list drive slots',
+            'list fans',
+            'list temp sensors',
+            'list voltage sensors',
+            'list current sensors',
+            'list power supplies',
+            'info enclosure'
+        ]
+
+        if self.get_option("debug"):
+            subcmds.append("getdebugcli")
+
+        result = self.collect_cmd_output(
+                    '%s list enclosures' % (cmd),
+                    suggest_filename='shmcli_list_enclosures')
+        if result['status'] == 0:
+            for line in result['output'].splitlines():
+                for model in models:
+                    if (model in line.split()):
+                        adapt_index = line.split()[-1]
+                        enc_index = line.split()[0]
+                        for subcmd in subcmds:
+                            if subcmd == 'getdebugcli':
+                                logpath = self.get_cmd_output_path(make=False)
+                                self.add_cmd_output(
+                                    '%s %s -a=%s -enc=%s -outputdir=%s'
+                                    % (cmd, subcmd,
+                                       adapt_index, enc_index, logpath),
+                                    suggest_filename="shmcli_%s_-a=%s \
+                                    -enc=%s_-outputdir=%s"
+                                    % (subcmd, adapt_index, enc_index,
+                                       logpath),
+                                    timeout=300)
+                            else:
+                                self.add_cmd_output(
+                                    '%s %s -a=%s -enc=%s'
+                                    % (cmd, subcmd, adapt_index, enc_index),
+                                    suggest_filename='shmcli_%s_-a=%s_-enc=%s'
+                                    % (subcmd, adapt_index, enc_index))
+
+        result = self.collect_cmd_output(
+                    '%s list drives' % (cmd),
+                    suggest_filename='shmcli_list_drives')
+        if result['status'] == 0:
+            for line in result['output'].splitlines():
+                if (len(line.split()) > 6):
+                    if (line.split()[0] not in ['WWN', '---']):
+                        wwn = line.split()[0]
```

Comment:
What does this output look like? We've had some trouble in the past with matching on line lengths, so I'd like to see if we might be able to do something a bit more robust here.

Reply:
Sure, however I was looking for like a pastebin or similar so I could play around with the output as written.